dds/idl/marshal_generator.cpp

```diff
@@ -3518,14 +3518,14 @@ bool marshal_generator::gen_union(AST_Union* node, UTL_ScopedName* name,
 
     // Add a reference to the idl file, if the descriminator is user defined.
     AST_Type* disc_type = resolveActualType(discriminator);
-    const Classification disc_cls = classify(disc_type);
+    const Classification disc_cls2 = classify(disc_type);
```

Comment:
Ideally the name of this variable would describe why it needs to exist as a distinct value from the similarly-named variable.  In this case I'm not sure it's needed at all.